app/controllers/UploadController.scala

```diff
@@ -0,0 +1,64 @@
+package controllers
+
+import java.util.UUID
+import javax.inject.Inject
+
+import com.amazonaws.services.securitytoken.model.AssumeRoleRequest
+import com.gu.pandahmac.HMACAuthActions
+import model.{UploadCredentials, UploadPolicy}
+import play.api.libs.json.{JsArray, JsObject, JsString, Json}
+import play.api.mvc.Results.Ok
+import util.{AWSConfig, Logging}
+
+class UploadController @Inject ()(implicit val authActions: HMACAuthActions, val awsConfig: AWSConfig) extends Logging {
+  val folder = "uploads"
+
+  import authActions.APIHMACAuthAction
+
+  def create(atomId: String) = APIHMACAuthAction {
+    log.info(s"Request for upload credentials for atom $atomId")
+
+    val uploadId = UUID.randomUUID().toString
```

Comment:
Yeah, depends how you define "same file". If you're just using filename then md5 is ok, but I think that's unlikely to be what you want.